### ISSUE TITLE
function parameter of type Literal[-1] gives incorrect error

### ISSUE THREAD

**Issue by judej** [microsoft/pylance-release#260]
## Environment data
Pylance language server 2020.8.1 (pyright 0916fb9e) 

```
def foo(neg1: Literal[-1]) -> None:
    pass

def bar(pos1: Literal[1]) -> None:
    pass

# the following results in error:
#     "int" cannot be assigned to type "Literal[-1]"Pylance 
foo(-1)

# this is ok
bar(1)
```

## Expected behaviour

No error for foo(-1) above

## Actual behaviour

"int" cannot be assigned to type "Literal[-1]"Pylance

**Comment by erictraut** [comment]
Thanks for the bug report. The type evaluator wasn't properly evaluating the type of expressions with unary +/- operators and integer literal operands. These should result in a literal type, but the literal value was being omitted when the unary expression was applied. I've fixed it in a way that +/- operations can be applied in a chained manner (e.g. "-(-++(-3))" evaluates to the type "Literal[-3]".

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.8.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202082-20-august-2020
